routes/auth/components/Signup.js

```diff
@@ -0,0 +1,130 @@
+const { useState, useRef } = require('react');
+const { useForm } = require('react-hook-form');
+const { Pressable } = require('react-native');
+const T = require('prop-types');
+const { default: Styled } = require('styled-components/native');
+const { Feather: Icons } = require('@expo/vector-icons');
+const { Button } = require('@ui-kitten/components');
+const { Link, Text } = require('../../../components/Type');
+const Input = require('../../../components/Form/Input');
+const IsEmail = require('../../../utils/is-email');
+const Theme = require('theme');
+
+const internals = {};
+
+module.exports = function Signup({ onSubmit, error }) {
+
+    const [canShowEmailError, setCanShowEmailError] = useState(false);
+    const [isSecureTextEntry, setIsSecureTextEntry] = useState(true);
+
+    const emailRef = useRef(null);
+    const passwordRef = useRef(null);
+
+    const { control, watch, handleSubmit, formState: { errors } } = useForm();
+    const fieldProps = { control, errors };
+
+    const toggleIsSecureEntry = () => setIsSecureTextEntry(!isSecureTextEntry);
+
+    const emailWatch = watch('email');
+    const passwordWatch = watch('password');
+
+    const renderIcon = (style, props) => {
+
+        const { tintColor, ...styleProps } = style;
+
+        return <Pressable onPress={toggleIsSecureEntry} hitSlop={Theme.spacing(2)}>
+            <Icons
+                {...props}
+                name={isSecureTextEntry ? 'eye' : 'eye-off'}
+                color={tintColor || Theme.palette.basic[800]}
+                size={20}
+                style={styleProps}
+            />
+        </Pressable>;
+    };
+
+    const renderCaption = () => {
+
+        const { CaptionContainer } = internals;
+
+        return (
+            <CaptionContainer>
+                <Icons
+                    name='alert-circle'
+                    color={Theme.palette.danger[400]}
+                    size={Theme.spacing(2)}
+                    style={{ marginRight: Theme.spacing(1) }}
+                />
+                <Text status='danger'>Invalid email address</Text>
+            </CaptionContainer>
+        );
+    };
```

Comment:
Haven't tried it yet but would we be able to maybe transfer these two functions to maybe an `internals.js` file and then maybe use them for both `Login` and `Signup` component?